### PR TITLE
Fix warning to duplicated spider. Issue 2181

### LINKED ISSUE #2181: rename spider will load the old left .pyc instead of renamed spider
Hi all, I'm not sure if this is an issue.
If got `y.py` and `y.pyc` in the `spider` directory.
Then rename it with `mv y.py x.py` and update the new `x.py`.
After updates, run `scrapy crawl spider` will use the old version of spider instead of updates in the `x.py`.
Have to remove `y.pyc` first.
Thanks!

### PR DESCRIPTION
**Pull request by MrMenezes**
Fixes #2181

As suggested, is notified every time there is more than one file with that name.

### CODE REVIEW

File: scrapy/spiderloader.py [hunk 1]
```diff
@@ -21,7 +21,12 @@ def __init__(self, settings):
             
     def _load_spiders(self, module):
         for spcls in iter_spider_classes(module):
-            self._spiders[spcls.name] = spcls
+            if spcls.name in self._spiders.keys():
+                import warnings
+                warnings.warn("There are several spiders with the same name (" + spcls.name +
+                              "), this can cause unexpected behavior", UserWarning)
+        self._spiders[spcls.name] = spcls
```
Comment (kmike) [inline scrapy/spiderloader.py]: this should be done in a loop, not after the loop

**Comment by kmike** [comment]
Thanks for the PR! Could you please add a test case for it? To check if a warning is raised you can use http://doc.pytest.org/en/latest/recwarn.html; SpiderLoader tests are here: https://github.com/scrapy/scrapy/tree/master/tests/test_spiderloader.

**Comment by redapple** [comment]
hi @MrMenezes , @kmike ,
I had a go at the testcase: https://github.com/MrMenezes/scrapy/pull/1
how does it look?

**Comment by redapple** [comment]
@MrMenezes , I took the liberty to submit my proposed changes on top of yours in #2612.
Hence I'm closing this PR. Thanks for bootstrapping this!

**Pull request closed without merging**